tests/core/test_linker.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+
+from eth_utils import is_address, remove_0x_prefix, to_hex
+import pytest
+
+from ethpm import Package
+from ethpm.utils.chains import get_chain_id
+from pytest_ethereum.deployer import Deployer
+from pytest_ethereum.exceptions import DeployerError, LinkerError
+from pytest_ethereum.linker import deploy, link
+from pytest_ethereum.utils.linker import contains_matching_uri, pluck_matching_uri
+
+MANIFEST_DIR = Path(__file__).parent.parent / "manifests"
+
+
+@pytest.fixture
+def escrow_deployer(solc_deployer, w3):
+    escrow_manifest_path = MANIFEST_DIR / "escrow_manifest.json"
+    return solc_deployer(escrow_manifest_path), w3
+
+
+@pytest.fixture
+def chain_setup(w3):
+    old_chain_id = remove_0x_prefix(to_hex(get_chain_id(w3)))
+    block_hash = remove_0x_prefix(to_hex(w3.eth.getBlock("earliest").hash))
+    old_chain_uri = "blockchain://{0}/block/{1}".format(old_chain_id, block_hash)
+    match_data = {
+        old_chain_uri: {"x": "x"},
+        "blockchain://{0}/block/{1}".format("1234", block_hash): {"x": "x"},
+    }
+    no_match_data = {
+        "blockchain://56775ac59d0774e6b603a79c4218efeb5653b99ba0ff14db983bac2662251a8a/block/{0}".format(  # noqa: E501
+            block_hash
+        ): {
+            "x": "x"
+        }
+    }
+    return w3, match_data, no_match_data, old_chain_uri
+
+
+def test_linker(escrow_deployer):
+    # todo test multiple links in one type
+    deployer, w3 = escrow_deployer
+    assert isinstance(deployer, Deployer)
+    with pytest.raises(DeployerError):
+        deployer.deploy("Escrow")
+
+    escrow_link_strategy = (
+        deploy("SafeSendLib"),
+        link("Escrow", "SafeSendLib"),
+        deploy("Escrow", w3.eth.accounts[0]),
+    )
+    deployer.register_strategy("Escrow", escrow_link_strategy)
+    escrow_deployer = deployer.deploy("Escrow")
+    linked_escrow_package, escrow_address = escrow_deployer
+    assert isinstance(linked_escrow_package, Package)
+    assert is_address(escrow_address)
+    linked_escrow_factory = linked_escrow_package.get_contract_factory("Escrow")
+    assert linked_escrow_factory.needs_bytecode_linking() is False
+
+
+def test_pluck_matching_uri(chain_setup):
```

Comment:
Mostly just an organizational thing, but I've found it is good to keep `utils` tests separate from the main tests.  Often once a utility is written you rarely have to update the tests, but the main functionality is likely to change more often.  By segregating them you get smaller test modules as well as the *utils* testing modules rarely being touched.